src/components/Game.jsx

```diff
@@ -0,0 +1,160 @@
+  const [isFetching, setIsFetching] = useState(false);
+  const [isGameWon, setIsGameWon] = useState(false);
+  const [canRevealAnswers, setCanRevealAnswers] = useState(false);
+  const [areAnswersRevealed, setAreAnswersRevealed] = useState(false);
+
+  const fetchQuestions = async () => {
+    setIsFetching(true);
+    try {
+      const response = await fetch(API_URL);
+      const responseData = await response.json();
+      return responseData.results;
+    } catch (error) {
+    } finally {
+      setIsFetching(false);
+    }
+  };
+
+  const selectAnswer = (currentQuestion, selectedAnswer) => {
+    setQuestionsData((prevState) =>
+      prevState.map((answer) =>
+        answer.question === currentQuestion
+          ? { ...answer, selectedAnswer }
+          : answer
+      )
+    );
+  };
+
+  const generateQuestionData = (rawQuestions) =>
+    rawQuestions.map((quizQuestion) => ({
+      question: quizQuestion.question,
+      selectedAnswer: null,
+      correctAnswer: quizQuestion.correct_answer,
+      allChoices: generateShuffledAnswers(
+        quizQuestion.incorrect_answers,
+        quizQuestion.correct_answer
+      ),
+    }));
+
+  const restartCurrentGame = () => {
+    setIsGameWon(false);
+    setScore(0);
+    setAreAnswersRevealed(false);
+    setCanRevealAnswers(false);
+    setQuestionsData((prevState) =>
+      prevState.map((question) => ({ ...question, selectedAnswer: null }))
+    );
+  };
+
+  const startNewGame = async () => {
+    const newQuestions = await fetchQuestions();
+    setQuestionsData(generateQuestionData(newQuestions));
+    setIsGameWon(false);
+    setScore(0);
+    setAreAnswersRevealed(false);
+    setCanRevealAnswers(false);
+  };
+
+  const handleAnswersReveal = () => {
+    setAreAnswersRevealed(true);
+    const score = questionsData.reduce((totalScore, currentQuestion) => {
+      if (currentQuestion.selectedAnswer === currentQuestion.correctAnswer)
+        return totalScore + 1;
+      return totalScore;
+    }, 0);
+    setScore(score);
+    if (score === questionsData.length) setIsGameWon(true);
+  };
+
+  useEffect(() => {
+    (async () => {
+      const quizQuestions = await fetchQuestions();
+      setQuestionsData(generateQuestionData(quizQuestions));
+    })();
+  }, []);
+
+  useEffect(() => {
+    const isAllQuestionsAnswered =
+      questionsData.length > 0 &&
+      questionsData.every((question) => question.selectedAnswer);
+    if (isAllQuestionsAnswered) setCanRevealAnswers(true);
+  }, [questionsData]);
+
+  return (
+    <section className="flex flex-col w-3/5 mx-auto mt-6 max-w-[1200px]">
+      {isFetching ? (
+        <h3 className="text-2xl font-semibold">Getting questions...</h3>
+      ) : (
+        <>
+          {isGameWon && (
+            <Confetti recycle={false} width={1200} className="mx-auto" />
+          )}
+
+          {questionsData.map((questionInfo, index) => {
+            const { question, allChoices, selectedAnswer, correctAnswer } =
```

Comment:
Here you seem to not be using the `questionInfo` variable anywhere else apart from here where you're destructuring. I'd recommend directly performing the destructuring to the argument like this: 
```js
questionsData.map(({ question, allChoices, selectedAnswer, correctAnswer }, index) => {}
```

Reply:
That looks more neat. Thanks!